File: Python/Lab4/TDate.py
import Funcs
from TTriad import TTriad


class TDate(TTriad):
    def __init__(self, number1, number2, number3):
        super().__init__(number1, number2, number3)

    def IncreaseNumber1(self, n):
        self.Number1 += n
        while self.Number1 > 30:
            self.Number1 -= 30
            self.IncreaseNumber2(1)

    def DecreaseNumber1(self, n):
        self.Number1 -= n
        while self.Number1 <= 0:
            self.Number1 += 30
            self.DecreaseNumber2(1)


    def IncreaseNumber2(self, n):
        self.Number2 += n
        while self.Number2 > 12:
            self.Number2 -= 12
            self.IncreaseNumber3(1)

    def DecreaseNumber2(self, n):
        self.Number2 -= n
        while self.Number2 <= 0:
            self.Number2 += 12
            self.DecreaseNumber3(1)


    def IncreaseNumber3(self, n):
        self.Number3 += n

    def DecreaseNumber3(self, n):
        self.Number3 -= n

    def Output(self):
        print(f"{Funcs.ChangeFormat(self.Number1)}.{Funcs.ChangeFormat(self.Number2)}.{Funcs.ChangeFormat(self.Number3)}")

File: C#/Lab4/TDate.cs
using System;

namespace lab4
{
    public class TDate : TTriad
    {
        public TDate(int n1, int n2, int n3) : base(n1, n2, n3)
        {
        }

        public override void IncreaseNumber1(int n)
        {
            Number1 += n;
            while (Number1 > 30)
            {
                Number1 -= 30;
                IncreaseNumber2(1);
            }
        }
        
        public override void DecreaseNumber1(int n)
        {
            Number1 -= n;
            while (Number1 <= 0)
            {
                Number1 += 30;
                DecreaseNumber2(1);
            }
        }
        
        public override void IncreaseNumber2(int n)
        {
            Number2 += n;
            while (Number2 > 12)
            {
                Number2 -= 12;
                IncreaseNumber3(1);
            }
        }
        
        public override void DecreaseNumber2(int n)
        {
            Number2 -= n;
            while (Number2 <= 0)
            {
                Number2 += 12;
                DecreaseNumber3(1);
            }
        }
        
        public override void IncreaseNumber3(int n)
        {
            Number3 += n;
        }

        public override void DecreaseNumber3(int n)
        {
            Number3 -= n;
        }

        public void Output()
        {
            Console.WriteLine($"{Funcs.ChangeFormat(Number1)}.{Funcs.ChangeFormat(Number2)}.{Funcs.ChangeFormat(Number3)}");
        }
    }
}